### PR TITLE
DynamoDB BatchGet does not work with more than 100 items

### PR DESCRIPTION
**Pull request by lucasas**
https://forums.aws.amazon.com/thread.jspa?threadID=100194

### CODE REVIEW

File: lib/dynamoid/adapter_plugin/aws_sdk_v2.rb [hunk 1]
```diff
@@ -38,38 +38,44 @@ def client
       #
       # @todo: Provide support for passing options to underlying batch_get_item http://docs.aws.amazon.com/sdkforruby/api/Aws/DynamoDB/Client.html#batch_get_item-instance_method
       def batch_get_item(table_ids, options = {})
-        request_items = Hash.new{|h, k| h[k] = []}
+request_items = Hash.new{|h, k| h[k] = []}
```
Comment (pboling) [inline lib/dynamoid/adapter_plugin/aws_sdk_v2.rb]: Please fix spacing.

File: lib/dynamoid/adapter_plugin/aws_sdk_v2.rb [hunk 2]
```diff
@@ -38,38 +38,44 @@ def client
...
           rng = tbl.range_key.to_s
 
-          keys = if rng.present?
-            Array(ids).map do |h,r|
-              { hk => h, rng => r }
-            end
-          else
-            Array(ids).map do |id|
-              { hk => id }
+          Array(ids).each_slice(100) do |ids|
```
Comment (pboling) [inline lib/dynamoid/adapter_plugin/aws_sdk_v2.rb]: Make `100` a configurable option in the `Config` class.  DynamoDB users will all have different paid settings, and some will need to set a smaller batch size.

**Comment by pboling** [comment]
Please rebase on latest master - Spec suite is fixed!!!

File: lib/dynamoid/adapter_plugin/aws_sdk_v2.rb (hunk 2 above)
Comment (peeyush1234) [inline lib/dynamoid/adapter_plugin/aws_sdk_v2.rb]: Making it configurable is a good idea. 
@pboling, What are your thoughts on using 100 as default if the batch size config is not provided by the client.

Comment (pboling) [inline lib/dynamoid/adapter_plugin/aws_sdk_v2.rb]: All I know about batch calls is there is a limit of 25 per batch on creates and deletes.  As this is for get, not create or delete, I am not sure what the default is, or the maximum.  If Amazon has a default batch size, or a recommended batch size we should use that.  If not then 100 seems reasonable.

**Comment by lucasas** [comment]
@pboling can you check if everything is ok?

**Comment by pboling** [comment]
Looks good!

**Pull request merged**